pkg/build/builder/cmd/scmauth/sshkey.go

```diff
@@ -21,10 +23,32 @@ func (_ SSHPrivateKey) Setup(baseDir string, context SCMAuthContext) error {
 	if err := script.Chmod(0711); err != nil {
 		return err
 	}
-	content := "#!/bin/sh\nssh -i " +
-		filepath.Join(baseDir, SSHPrivateKeyMethodName) +
-		" -o StrictHostKeyChecking=false \"$@\"\n"
-
+	foundPrivateKey := false
+	foundKnownHosts := false
+	files, err := ioutil.ReadDir(baseDir)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		switch {
+		case file.Name() == knownHostsFileName:
+			foundKnownHosts = true
+		case file.Name() == SSHPrivateKeyMethodName:
+			foundPrivateKey = true
+		}
+		log.V(5).Infof("source secret dir %s has file %s", baseDir, file.Name())
+	}
+	if !foundPrivateKey {
+		return fmt.Errorf("could not find the ssh-privatekey file for the ssh secret stored at %s", baseDir)
+	}
+	// let's see if known_hosts was included in the secret
+	knownHostsPath := filepath.Join(baseDir, knownHostsFileName)
```

Comment:
this could be moved down into the else block since that's the only path where it's used.

Reply:
update to move down pushed thanks